cfme/common/vm.py

```diff
@@ -606,32 +606,32 @@ def is_pwr_option_available_in_cfme(self, option, from_details=False):
             return False
 
     def delete_from_provider(self):
-        logger.info("Begin delete_from_provider")
-        if self.provider.mgmt.does_vm_exist(self.name):
-            # InfraProviders require VM to be stopped before removal
-            if self.provider.one_of(InfraProvider):
-                try:
-                    if self.provider.mgmt.is_vm_suspended(self.name):
-                        logger.debug("Powering up VM %s to shut it down correctly on %s.",
-                                    self.name, self.provider.key)
-                        self.provider.mgmt.start_vm(self.name)
-                        self.provider.mgmt.wait_vm_steady(self.name)
-                        self.provider.mgmt.stop_vm(self.name)
-                        self.provider.mgmt.wait_vm_steady(self.name)
-                except exceptions.ActionNotSupported:
-                    # Action is not supported on mgmt system. Simply continue
-                    pass
-            # One more check (for the suspended one)
-            if self.provider.mgmt.does_vm_exist(self.name):
-                try:
-                    logger.info("Mgmt System delete_vm")
-                    return self.provider.mgmt.delete_vm(self.name)
-                except exceptions.VMInstanceNotFound:
-                    # Does not exist already
-                    return True
-        else:
+        """
+        Delete VM/instance from provider.
+
+        You cannot expect additional cleanup of attached resources by calling this method. You
+        should use cleanup_on_provider() to guarantee that.
+        """
+        logger.info("Begin delete_from_provider for VM '{}'".format(self.name))
+        if not self.provider.mgmt.does_vm_exist(self.name):
+            logger.info("VM does '{}' not exist on provider, nothing to delete".format(self.name))
             return True
 
+        # Some providers require VM to be stopped before removal
+        logger.info(
+            "delete: ensuring VM '{}' on provider '{}' is powered off".format(
+                self.name, self.provider.key)
+        )
+        self.ensure_state_on_provider(self.STATE_OFF)
+
+        logger.info("delete: removing VM '{}'".format(self.name))
+        try:
+            return self.provider.mgmt.delete_vm(self.name)
+        except Exception:
+            logger.exception("delete for vm '{}' failed".format(self.name))
+
+        return True
```

Comment:
**optional** based on skimming the code i believe the return value is unchecked to begin with, please evaluate if it should be created/given at all